plugin/code_actions.py

```diff
@@ -119,7 +119,10 @@ def _request_async(
         only_with_diagnostics: bool,
         actions_handler: Callable[[CodeActionsByConfigName], None],
         on_save_actions: Optional[Dict[str, bool]] = None
-    ) -> CodeActionsCollector:
+    ) -> None:
+        if 'codeActionProvider' in userprefs().disabled_capabilities:
+            return None
```

Comment:
If we keep this logic in here then at least `actions_handler` should be called with empty results instead of returning `None`. (Also can be just `return` instead of `return None`)

Reply:
And BTW. The callback is called on async thread so we should be consistent here.

Reply:
Good point about actions_handler. Shows another useful property of promises/futures as you wouldn't forget to return a resolved promise due to the type system.